products/messaging/frontend/Campaigns/Campaign.tsx

```diff
@@ -0,0 +1,90 @@
+import '@xyflow/react/dist/style.css'
+
+import { LemonTabs } from '@posthog/lemon-ui'
+import { useValues } from 'kea'
+import { Form } from 'kea-forms'
+import { router } from 'kea-router'
+import { PageHeader } from 'lib/components/PageHeader'
+import { LemonField } from 'lib/lemon-ui/LemonField'
+import { LemonInput } from 'lib/lemon-ui/LemonInput'
+import { capitalizeFirstLetter } from 'lib/utils'
+import { SceneExport } from 'scenes/sceneTypes'
+import { urls } from 'scenes/urls'
+
+import { campaignLogic, CampaignLogicProps, CampaignTab, CampaignTabs } from './campaignLogic'
```

Comment:
Recommendation - set up a "campaignSceneLogic" which _just_ takes care of the top level routing. 

Didn't do this for HogFunctions and it made it super hard to re-use the logic later.